### PR TITLE
[Workflow] remove new lines from workflow metadata

### LINKED ISSUE #49174: [Workflow] New lines in the label of a transition break the PUML file
### Symfony version(s) affected

5.4+

### Description

I tried to add new lines in the label of a transition, in order to have a cleaner output in the PUML file (then transformed to PNG by our toolchain).

So I instinctively wrote this:

```yaml
                        label: |
                            unpublish
                            (Lorem ipsum dolor sit amet)
```

It didn't added some new lines, which is an acceptable behaviour.

The problem is that the resulting PUML file had an invalid syntax, and the PUML parser (a `think/plantuml` Docker image) returns “Syntax error?”.

### How to reproduce

```yaml
# config/packages/workflow.yaml
framework:
    workflows:
        my_workflow:
            transitions:
                unpublish:
                    from: [published]
                    to: pending
                    metadata:
                        arrow_color: 'Grey'
                        label: |
                            unpublish
                            (Lorem ipsum dolor sit amet)
```

### Possible Solution

Remove new lines from the labels.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by alexislefebvre**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #49174
| License       | MIT
| Doc PR        | no

Don't dump new lines in workflow dumps.

### CODE REVIEW

File: src/Symfony/Component/Workflow/Dumper/PlantUmlDumper.php [hunk 1]
```diff
@@ -246,4 +248,13 @@ private function getColorId(string $color): string
         // Remove “#“ from start of the color name so it can be used as an identifier.
         return ltrim($color, '#');
     }
+
+    private static function removeNewLines(?string $subject): ?string
```
Comment (OskarStark) [inline src/Symfony/Component/Workflow/Dumper/PlantUmlDumper.php]: Shall we move this to a trait? putting this into an extra class seems a bit overkill to me

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/Workflow/Dumper/MermaidDumper.php [hunk 2]
```diff
@@ -285,4 +285,13 @@ private function styleLink(array $transitionMeta): string
 
         return '';
     }
+
+    private static function removeNewLines(?string $subject): ?string
```
Comment (lyrixx) [inline src/Symfony/Component/Workflow/Dumper/MermaidDumper.php]: I would call this method `sanitize`: if we need to escape / replace / sanitize it'll be a good entry point! WDYT?

**Comment by alexisLefebvre** [comment]
Thanks for the reviews, suggestions have been taken into account.

**Review by OskarStark — APPROVED** [review]

**Comment by lyrixx** [comment]
I forgot to ask, are you sur other dumper needs this patch?

**Comment by alexisLefebvre** [comment]
I used the sanitizer so that the tests are not broken but I don't know if Mermaid or Graphviz accept new lines.

**Comment by lyrixx** [comment]
IMHO, you should fix only what is broken ; so there is not need for the trait anymore I guess.

Sorrry to go back and forth!

**Comment by alexisLefebvre** [comment]
Fair enough, let's not remove new lines if the tools support them, I'll check later.

Status: Needs work

**Comment by alexisLefebvre** [comment]
It took some examples and added new lines in the transitions.

## Graphviz Dot

With this input: https://github.com/symfony/symfony/blob/bc5fea4e2190f00a207871115a83dd7df03d7637/src/Symfony/Component/Workflow/Tests/Dumper/StateMachineGraphvizDumperTest.php#L62

Passing it to the Docker image `nshine/dot` gives this output:

> dot

New lines don't cause errors, and they appear in the result. :heavy_check_mark: 

## Mermaid

Adding a new line in any transition in the [live editor](https://mermaid.live/edit) results in a Parse error.

I'm pretty sure that the live editor behaves like Mermaid, so I didn't test further. :negative_squared_cross_mark: 

## Summary

Symfony can keep new lines in the Graphviz dumper. The other formats need new lines to be removed.

**Comment by alexisLefebvre** [comment]
Status: Needs review

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/Workflow/Dumper/DumperHelperTrait.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Dumper;
+
+trait DumperHelperTrait
```
Comment (fabpot) [inline src/Symfony/Component/Workflow/Dumper/DumperHelperTrait.php]: I would remove this trait and inline the code.

Comment (alexisLefebvre) [reply src/Symfony/Component/Workflow/Dumper/DumperHelperTrait.php]: I removed it.

**Review by lyrixx — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @alexislefebvre.

**Pull request merged**